### PR TITLE
feat: Handle out of band seeks

### PR DESCRIPTION
**Pull request by tmdiep**
Handles `RESET` signal (out of band seek notification) from the server by:

* Discarding undelivered messages. Discarding outstanding acks for delivered messages.
* Waiting for the committer to flush pending commits and receive the acknowledgment from the server.
* Resetting all committer and subscriber state.

Sets the `InitialSubscribeRequest.initial_location` field when reconnecting subscribe streams.

### CODE REVIEW

File: google/cloud/pubsublite/cloudpubsub/internal/single_partition_subscriber.py [hunk 1]
```diff
@@ -75,13 +90,14 @@ async def read(self) -> Message:
             )
             cps_message = self._transformer.transform(message)
             offset = message.cursor.offset
+            ack_id = f"{self._ack_generation_id}/{offset}"
```
Comment (dpcollins-google) [inline google/cloud/pubsublite/cloudpubsub/internal/single_partition_subscriber.py:93]: can you instead break this out into a simple class with parse(payload: str) and __str__ methods? You can see MessageMetadata as a template for this.

Comment (dpcollins-google) [inline google/cloud/pubsublite/cloudpubsub/internal/single_partition_subscriber.py:93]: can you instead break this out into a simple class with parse(payload: str) and __str__ methods? You can see MessageMetadata as a template for this.

**Comment by dpcollins-google** [comment]
Approved contingent on fixes.

**Review by dpcollins-google — APPROVED** [review]

File: google/cloud/pubsublite/internal/wire/subscriber_impl.py [hunk 2]
```diff
@@ -152,7 +160,30 @@ async def reinitialize(
     ):
         self._reinitializing = True
         await self._stop_loopers()
-        await connection.write(SubscribeRequest(initial=self._initial))
+        if last_error and is_reset_signal(last_error):
+            # Discard undelivered messages and refill flow control tokens.
+            while True:
```
[the remark this answers is not in the document]
Comment (tmdiep) [reply google/cloud/pubsublite/internal/wire/subscriber_impl.py]: Done

File: google/cloud/pubsublite/cloudpubsub/internal/single_partition_subscriber.py (hunk 1 above)
Comment (tmdiep) [reply google/cloud/pubsublite/cloudpubsub/internal/single_partition_subscriber.py]: Done

Comment (tmdiep) [reply google/cloud/pubsublite/cloudpubsub/internal/single_partition_subscriber.py]: Added a private `_AckId` class. LMK if you intended this to be user accessible.

**Pull request merged**